### ISSUE TITLE
[FR] Fixed position proposal for the styles preview in the lighttable.

### ISSUE THREAD

**Issue by difrkaguilar** [darktable-org/darktable#12872]
**Is your feature request related to a problem? Please describe.**

Looking at the recent addition of the style preview in the styles module of the lighttable, great BTW, I saw that the preview pop-up window changes size depending on the number of modules applied to the image development that have been added as a style. The window changes back and forth indifferently showing these modules.

**Describe the solution you'd like**

It would be good to achieve greater consistency in the preview of the styles, to have fixed the position of this window, where what changes is the image horizontally or vertically and use the icon already recognized by users to show the number of modules applied as part of the style, this icon is present both in the light table view and in the previews of the culling layout. Only if users place the mouse pointer over the icon, the pop-up menu will be displayed.

In that case it won't be necessary to repeat the style name as it is now in the pop-up window. And the preview will be resizable using the Ctrl+scroll shortcut as the graphs in curves or filmic.

Of course, it would be necessary to find another solution to the preview of the styles in the darkroom, since the same thing happens there, the pop-up menu shows the styles, and the window changes sizes with their corresponding modules applied and saved as styles.

**reference images**

![Screenshot from 2022-11-11 19-04-30](https://user-images.githubusercontent.com/67672868/201451510-d0637a94-ef01-4282-815c-5ff1c2e73261.png)
![Screenshot from 2022-11-11 19-04-53](https://user-images.githubusercontent.com/67672868/201451516-cd12c8e9-4abc-4f4d-88c9-b066ad7c4bd0.png)
![Screenshot from 2022-11-11 19-05-19](https://user-images.githubusercontent.com/67672868/201451528-1235a828-89da-4a08-a821-bdb500842f7d.png)

icon used in the culling layout with the popup menu.

![Screenshot from 2022-11-11 17-54-25](https://user-images.githubusercontent.com/67672868/201452097-e115b7b0-154f-412f-b5cf-de13f3487dc3.png)

**mockup proposal**

version with fixed 1x1 box with the image inside in horizontal or vertical position and the icon to show the used modules in the current selected style.

![Style_window_v01](https://user-images.githubusercontent.com/67672868/201451600-db51049f-8d2e-45f7-bf3d-b1cbc63a759b.png)

version with image taking the biggest area possible in horizontal or vertical position and the icon to show the used modules in the current selected style.

![Style_window_v02](https://user-images.githubusercontent.com/67672868/201451896-cc081c2b-5614-4f63-ba5d-6852eba2ab3d.png)

@TurboGit @Nilvus PS. Sorry for my useless English. :-)

**Comment by TurboGit** [comment]
I'll be more in favor of keep this as a tooltip to avoid loosing too much vertical space on the style module. But I don't know how to position a tooltip in absolute coordinate to have it always at the same place. Then, we could add the image before the module list to ensure the image will always be at the same place.

If a Gtk expert read that, can you please look at this or give pointers to how position a tooltip Window?

**Comment by TurboGit** [comment]
@dterrahe : Maybe you know how to do that?

**Comment by TurboGit** [comment]
@AlicVB : You are also a good candidate to help here. Maybe reworking as a popup?

**Comment by dterrahe** [comment]
I haven't looked at this yet, sorry. I will later this afternoon.

**Comment by AlicVB** [comment]
yep, popup can be a solution to finetune the position, but not for the size. But tbh, the size will differ in any case and I'm not sure it's worth the pain to handle this...
For me the current situation is completely ok. The fact that what we show is a tooltip is perfectly clear, and user may be used to the fact that tooltips window sizes change depending of its content.
What may (or not) be in favor of a popup is more that this will allow to show the window without any delay...

btw, I agree with @TurboGit about the "precious" vertical size (that plus the fact that this won't be possible to achieve same UI in darkroom)

**Comment by dterrahe** [comment]
> popup can be a solution to finetune the position

agreed (popover)

> but not for the size

Why not?

**Comment by AlicVB** [comment]
> > but not for the size
> 
> Why not?

because we can't know in advance the nb of line of the style history... so the height at least will change for each style (or I may have misunderstood something)

**Comment by dterrahe** [comment]
I thought we were talking about the size of the preview image?

So I guess the problem is in the light table if we anchor the popover to the module (header or somewhere) it might go up or down depending on how much space it needs and is available. Whereas in the darkroom, we anchor it to the button which is at the bottom so it will always (?) go up, so we put the image at the bottom of the popover so it doesn't have to move no matter how many lines are above it).
